File: modules/student-experience/src/main/java/com/campusos/portal/service/Scope.java
package com.campusos.portal.service;

/** Both ids or nothing. Every service and repository call carries this. */
public record Scope(String tenantId, String studentId) {
    public Scope {
        if (tenantId == null || tenantId.isBlank())
            throw new IllegalStateException("tenantId missing — query refused");
        if (studentId == null || studentId.isBlank())
            throw new IllegalStateException("studentId missing — query refused");
    }
}

File: modules/student-experience-js/src/service/Scope.js
/** Both ids or nothing. Every service and repository call carries this. */
export class Scope {
  constructor(tenantId, studentId) {
    if (!tenantId || !tenantId.trim()) throw new Error('tenantId missing — query refused');
    if (!studentId || !studentId.trim()) throw new Error('studentId missing — query refused');
    this.tenantId = tenantId;
    this.studentId = studentId;
    Object.freeze(this);
  }
}
